### PR TITLE
Obmin 285 Migration from Springfox Swagger UI to Springdoc Swagger UI

### PR DESCRIPTION
**Pull request by ennnbaev**

### CODE REVIEW

**Review by IllyaZuew — APPROVED** [review]

File: src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java [hunk 1]
```diff
@@ -1,60 +1,45 @@
...
-    }
-
-    private ApiInfo apiInfo() {
-        return new ApiInfoBuilder()
-                .title("Obminyashka (Child Goods Exchange) API")
-                .description("API Definitions of the Obminyashka (Child Goods Exchange) project")
-                .version("1.0.0")
+    public GroupedOpenApi publicApi() {
+        return GroupedOpenApi.builder()
+                .group("springshop-public")
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java:20]: You should construct the API description with the values we had previously here.

File: src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java [hunk 2]
```diff
@@ -1,60 +1,45 @@
...
-                .securityReferences(defaultAuth())
-                .build();
-    }
-
-    private List<SecurityReference> defaultAuth() {
-        return List.of(new SecurityReference(HttpHeaders.AUTHORIZATION,
-                new AuthorizationScope[]{
-                        new AuthorizationScope("global", "accessEverything")}));
+    @Bean
+    public OpenAPI springShopOpenAPI() {
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java:26]: The name of the method has the example name and should be renamed for our needs.

Comment (ennnbaev) [reply src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java:26]: Something like that: api or obminyashkaOpenAPI?

File: src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java [hunk 3]
```diff
@@ -1,60 +1,45 @@
...
+        final String securitySchemeName = "Authorization";
```
[the remark this answers is not in the document]
Comment (ennnbaev) [reply src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java:27]: In springdoc we don't have this class

**Review by ImAndVlad — APPROVED** [review]

**Review by rpkyrych — APPROVED** [review]

**Pull request merged**